config/entry.go

```diff
@@ -34,19 +34,19 @@ func makeEntryFromValue(value any) *Entry {
 }
 
 func (e *Entry) validate(key string) error {
-	if e.Value == nil {
-		if e.Required {
-			return errors.Errorf("%q is required and must contain a value", key)
-		}
-		// When just nil, we don't need to validate.
-		return nil
-	}
-
 	if err := e.tryEnvVarConversion(key); err != nil {
 		return err
 	}
 
+	v := reflect.ValueOf(e.Value)
+	if e.Required && (!v.IsValid() || e.Value == nil) {
```

Comment:
```suggestion
	if e.Required && (!v.IsValid() || v.IsNil()) {
```

Reply:
You're right. Here is what I suggest to cover the scenario I showed in the playground link earlier.
```suggestion
	if e.Required && (!v.IsValid() || e.Value == nil || (v.Kind() == reflect.Pointer && v.IsNil())) {
```